examples/artifacts/plot_detect_riemannian_potato_EEG.py

```diff
@@ -0,0 +1,248 @@
+from mne.io import read_raw_fif
+from mne import Epochs, make_fixed_length_events
+from pyriemann.estimation import Covariances
+from pyriemann.clustering import Potato
+from matplotlib import pyplot as plt
+from matplotlib.colors import to_rgb
+from matplotlib.animation import FuncAnimation
+
+
+###############################################################################
+
+
+@partial(np.vectorize, excluded=['potato'])
+def get_zscores(cov_00, cov_01, cov_11, potato):
+    cov = np.array([[cov_00, cov_01], [cov_01, cov_11]])
+    with np.testing.suppress_warnings() as sup:
+        sup.filter(RuntimeWarning)
+        return potato.transform(cov[np.newaxis, ...])
+
+
+def plot_potato_2D(ax, cax, X, Y, p_zscores, p_center, covs, p_colors, clabel):
+    qcs = ax.contourf(X, Y, p_zscores, levels=20, vmin=p_zscores.min(),
+                      vmax=p_zscores.max(), cmap='RdYlBu_r', alpha=0.5)
+    ax.contour(X, Y, p_zscores, levels=[z_th], colors='k')
+    sc = ax.scatter(covs[:, 0, 0], covs[:, 1, 1], c=p_colors)
+    ax.scatter(p_center[0, 0], p_center[1, 1], c='k', s=100)
+    if cax:
+        cbar = fig.colorbar(qcs, cax=cax)
+    else:
+        cbar = fig.colorbar(qcs, ax=ax)
+    cbar.ax.set_ylabel(clabel)
+    return sc
+
+
+def plot_sig(ax, time, sig):
+    ax.axis((time[0], time[-1], -15, 15))
+    pl, = ax.plot(time, sig, lw=0.75)
+    ax.axhspan(
+        -14, 14, edgecolor='k', facecolor='none',
+        xmin=-test_time_start / (test_time_end - test_time_start),
+        xmax=(duration - test_time_start) / (test_time_end - test_time_start))
+    return pl
+
+
+def add_alpha(p_cols, alphas):
+    cols = [to_rgb(c) for c in p_cols]
+    return [(c[0], c[1], c[2], a) for c, a in zip(cols, alphas[-len(p_cols):])]
+
+
+###############################################################################
+# Load EEG data
+# -------------
+
+raw_fname = os.path.join(sample.data_path(), 'MEG', 'sample',
+                         'sample_audvis_filt-0-40_raw.fif')
+raw = read_raw_fif(raw_fname, preload=True, verbose=False)
+sfreq = int(raw.info['sfreq'])
+
+
+###############################################################################
+# Offline processing of EEG data
+# ------------------------------
+
+# Apply common average reference on EEG channels
+raw.pick_types(meg=False, eeg=True).apply_proj()
+
+# Select two EEG channels for the example, preferably without artifact at the
+# beginning to have a reliable calibration
+ch_names = ['EEG 010', 'EEG 015']
+
+# Apply band-pass filter between 1 and 35 Hz
+raw.filter(1., 35., method='iir', picks=ch_names)
+
+# Epoch time-series with a sliding window
+duration = 2.5    # duration of epochs
+interval = 0.2    # interval between epochs
+epochs = Epochs(raw, make_fixed_length_events(raw, id=1, duration=interval),
+                tmin=0., tmax=duration, baseline=None, verbose=False)
+epochs_data = 5e5 * epochs.get_data(picks=ch_names)
```

Comment:
you can now use mne.make_fixed_length_epochs

https://mne.tools/dev/generated/mne.make_fixed_length_epochs.html

Reply:
```
""" Test make_fixed_length_events vs make_fixed_length_epochs """

import os
from matplotlib import pyplot as plt
from mne.datasets import sample
from mne.io import read_raw_fif
from mne import Epochs, make_fixed_length_events, make_fixed_length_epochs

###############################################################################
# Load EEG data
# -------------

raw_fname = os.path.join(sample.data_path(), 'MEG', 'sample',
                         'sample_audvis_filt-0-40_raw.fif')
raw = read_raw_fif(raw_fname, preload=True, verbose=False)
sfreq = int(raw.info['sfreq']) # 150 Hz

# Keep 2 EEG channels
raw.pick_types(meg=False, eeg=True).pick_channels(['EEG 001', 'EEG 010'])
raw.filter(1., 35.)
print(raw.info)
raw_shape = raw.get_data().shape 
# shape = 2 channels and 41700 samples (or 278 seconds)

# Epoch EEG time-series with a sliding window
length = 2.5      # length of window = length of epochs
interval = 0.2    # interval / overlap between consecutive epochs

# So, we must approximatively (without taking account side effects)
# obtain: 278/0.2 = 1390 epochs

###############################################################################
# Solution 1: define an event for each epoch, then use Epoch
# ----------------------------------------------------------

# make_fixed_length_events: make a set of events separated by a fixed duration.
# In our case, overlaped epochs are separated by a duration equal to interval.
events = make_fixed_length_events(raw, id=1, duration=interval)
print("shape of events =", events.shape)
# shape = (1388, 3) => OK

# Plot events on EEG raw signal
fig_r = raw.plot(events=events, duration=8, start=0)
plt.savefig("fig_r")

# Epoch EEG on events, from 0 to 2.5 seconds
epochs1 = Epochs(raw, events, tmin=0., tmax=length, baseline=None)
print("shape of epochs1 =", epochs1.get_data().shape)
# shape = (1377, 2, 376) => OK

# Plot epochs1
fig_e1 = epochs1.plot(n_epochs=5)
plt.savefig("fig_e1")

###############################################################################
# Solution 2: use make_fixed_length_epochs
# ----------------------------------------

# make_fixed_length_epochs: epoch data into equal-sized consecutive epochs.
# duration: duration of each epoch in seconds = length
# overlap: the overlap between epochs, in seconds = interval
epochs2 = make_fixed_length_epochs(raw, duration=length, overlap=interval,
                                   reject_by_annotation=False)
print("shape of epochs2 =", epochs2.get_data().shape)
# shape = (120, 2, 375) => not the good number of epochs => KO

# Plot epochs2
fig_e2 = epochs2.plot(n_epochs=5)
plt.savefig("fig_e2")
# Epochs are not overlapped => KO
```